static/analysis.go

```diff
@@ -1,32 +1,43 @@
 package static
 
 import (
+	"fmt"
 	"go/constant"
 	"go/types"
 
 	"github.com/dogmatiq/configkit"
 	"github.com/dogmatiq/configkit/internal/entity"
 	"github.com/dogmatiq/configkit/internal/typename/gotypes"
+	"github.com/dogmatiq/configkit/message"
 	"golang.org/x/tools/go/ssa"
 )
 
 // analyzeApplication analyzes a type that implements the dogma.Application
-// interface to deduce it's application configuration.
-func analyzeApplication(pkg *ssa.Package, typ types.Type) configkit.Application {
+// interface to deduce its application configuration.
+func analyzeApplication(prog *ssa.Program, typ types.Type) configkit.Application {
 	app := &entity.Application{
-		TypeNameValue: gotypes.NameOf(typ),
+		TypeNameValue:     gotypes.NameOf(typ),
+		HandlersValue:     configkit.HandlerSet{},
+		MessageNamesValue: configkit.EntityMessageNames{},
 	}
 
-	fn := pkg.Prog.LookupMethod(
-		typ,
-		pkg.Pkg,
-		"Configure",
-	)
+	pkg := getTypePkg(typ)
+	fn := prog.LookupMethod(typ, pkg, "Configure")
 
 	for _, c := range findConfigurerCalls(fn) {
+		args := c.Common().Args
+
 		switch c.Common().Method.Name() {
 		case "Identity":
 			app.IdentityValue = analyzeIdentityCall(c)
+		case "RegisterAggregate":
+			app.HandlersValue.Add(
+				analyzeHandler(
```

Comment:
What happens if the type passed to RegisterAggregate doesn't implement `dogma.AggregateMessageHandler`, I guess compilation would have failed and resulted in a `nil` package earlier?

Reply:
Correct, this will be a typical package error that would result in `nil` package as an output from the `ssa.LoadPackages()` function.

Reply:
Ok, cool!